public/docs/dart/latest/guide/dependency-injection.jade

```diff
@@ -1,279 +1,138 @@
-  [Non-class dependencies](#non-class-dependencies) and
-  [OpaqueToken](#opaquetoken) sections.
+  .callout.is-helpful
+    header Dart difference: Interfaces are valid tokens
+    :marked
+      In TypeScript, interfaces don't work as provider tokens.
+      Dart doesn't have this limitation;
+      every class implicitly defines an interface,
+      so interface names are just class names.
 
-+includeShared('{ts}', 'providers-factory-1')
-+makeExample('dependency-injection/dart/lib/heroes/hero_service.dart','internals', 'lib/heroes/hero_service.dart (excerpt)')(format='.')
-+includeShared('{ts}', 'providers-factory-2')
-+makeExample('dependency-injection/dart/lib/heroes/hero_service_provider.dart','factory', 'lib/heroes/hero_service_provider.dart (excerpt)')(format='.')
-+includeShared('{ts}', 'providers-factory-3')
-+makeExample('dependency-injection/dart/lib/heroes/hero_service_provider.dart','provider', 'lib/heroes/hero_service_provider.dart (excerpt)')(format='.')
-+includeShared('{ts}', 'providers-factory-4')
-+includeShared('{ts}', 'providers-factory-5')
-- var stylePattern = { otl: /(providers.*),$/gm };
-+makeTabs(
-  `dependency-injection/dart/lib/heroes/heroes_component.dart,
-  dependency-injection/dart/lib/heroes/heroes_component_1.dart`,
-  null,
-  `lib/heroes/heroes_component (v.3),
-  lib/heroes/heroes_component (v.2)`,
-  stylePattern)
-+includeShared('{ts}', 'tokens-1')
-+makeExample('dependency-injection/dart/lib/injector_component.dart','get-hero-service')(format='.')
-+includeShared('{ts}', 'tokens-2')
-// [PENDING: How about a better name than ProviderComponent8?]
-+makeExample('dependency-injection/dart/lib/providers_component.dart','provider-8-ctor')(format=".")
-+includeShared('{ts}', 'tokens-3')
-.callout.is-helpful
-  header Dart difference: Interfaces are valid tokens
   :marked
-    In TypeScript, interfaces don't work as provider tokens.
-    Dart doesn't have this problem;
-    every class implicitly defines an interface,
-    so interface names are just class names.
-+includeShared('{ts}', 'tokens-non-class-deps-1')
-:marked
-  We know we can register an object with a [value provider](#value-provider).
-  But what do we use for the token?
-+includeShared('{ts}', 'tokens-opaque-1')
-+makeExample('dependency-injection/dart/lib/providers_component.dart','opaque-token')(format='.')
-+includeShared('{ts}', 'tokens-opaque-2')
-+makeExample('dependency-injection/dart/lib/providers_component.dart','use-opaque-token')(format=".")
-:marked
-  Here's an example of providing configuration information
-  for an injected class. First define the class:
-+makeExample('dependency-injection/dart/lib/providers_component.dart','configurable-logger')(format=".")
-:marked
-  Then inject that class and its configuration information:
-+makeExample('dependency-injection/dart/lib/providers_component.dart','providers-7')(format=".")
+    The problem with using `Map` as a token, is not that it is an abstract
```

Comment:
This is a really long sentence, and it continues the thought of the previous paragraph. Let's shorten this sentence and combine the two paragraphs. Maybe:

But what should we use as the token? We can't use `Map` because it's too general purpose. Our app might depend on several maps, each for a different purpose.

Dart difference: Interfaces are valid tokens
... are just class names. `Map` is a _valid_ token even though it's the name of an abstract class; it's just _unsuitable_ as a token because it's too general.